### ISSUE TITLE
Enable bike checkin/checkout for the West Corridor bike library

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#633]
This will not be merged to master.
- Add the QR code plugin
- Add a tab to checkin/checkout
  - add checkin/checkout buttons
  - add a display of the currently checked out bike
- add a new collection to the DB
- add a new API call to checkin/checkout
- update the collection with the bike/person mapping

**Comment by shankari** [comment]
The first two tasks are now complete and the UI update has been pushed.

wrt the server side changes, we can just add a new collection to the DB, and/or we can actually create a new data type and store an entry for the user each time they checkin or checkout the bikes. I think we will still need the collection for efficient retrieval of the current list of checked out bikes.

We can't do this as part of post-processing since the state can then be inconsistent for a long time, which defeats the purpose of having it in the first place.

If we ever changed to running the pipeline on demand when we pushed data, *and* we pushed the entry synchronously, we could solve this. But changing the pipeline to synchronous operation is a bigger lift than I am willing to tackle at this point. And it is not clear if this is ever going to become a core feature. So let's just create the new collection.
